_posts/ericabrody/2016-02-03-Erica-Brody-Ch5-Exercises.md

```diff
@@ -0,0 +1,18 @@
+---
+layout: post
+author: ericabrody
+title: "Erica's Chapter 3 Exercises"
+---
+
+Exercise 1: Write a program which repeatedly reads numbers until the user enters “done”. Once “done” is entered, print out the total, count, and average of the numbers. If the user enters anything other than a number, detect their mistake using try and except and print an error message and skip to the next number.
+<iframe src="https://trinket.io/embed/python/e0f9320136" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
```

Comment:
Overall, this program is great.  One edge case to consider: if the user presses 'enter' without inputting anything the averages etc become `nan`, which stands for not a number.  This is a case where `try` `except` has failed you because `float()` doesn't throw an actual error.  You could potentially test if the user input was nothing and `continue`.

That said, there will always be edge cases :)